Documentation: Update formatting and structure for Development environment document

bisko opened the pull request:
#### Changes proposed in this Pull Request:

* This PR introduces an update to the Development Environment setup document.
* Updated:
 * Formatting
  * Some things were a bit hard to follow when I set up my environment
 * Organization
  * Some points are now grouped together 
  * Tried to order the document in more step-by-step flow, compared to more "group of notes" style

#### Testing instructions:

* Open `docs/development-environment.md`
* Read the document and see if it makes sense and does it look better than it did before

#### Proposed changelog entry for your changes:

* Not sure if this will need a changelog entry as it doesn't affect the plugin itself.

bisko committed:
Initial commit - improve development environment documentation.

Started updating the main development environment document to make it a bit more organized and reduce clutter.

Added table of contents and set general outline for the document.

bisko committed:
More updates to the documentation, still WIP

bisko committed:
More formatting and updates

bisko committed:
More updates to the documentation and formatting

bisko committed:
More formatting updates

bisko committed:
Remove empty and unneeded file that was added previously

leogermani commented on docs/development-environment.md:
```diff
@@ -1,298 +1,444 @@
+## PHP unit tests
 
-## Installing Composer
+* ### Docker
```
The re-organization looks great to me. My only question is whether we should add this here or leave it just in the Docker page...

bisko replied:
I think having the section here is good. Now that you mention it though, I'm not sure if we should be duplicating instructions between both documents, in case they change. For the moment I think this would be fine and should be updated in future iterations on the documentation.

jeherve commented on docs/development-environment.md:
````diff
@@ -1,298 +1,444 @@
-phpunit -c tests/php.multisite.xml
-```
+<a name="jetpack-development-build"></a>
+## Building Jetpack
````
I wonder if we should take the opportunity to split things up a bit here. We now have a build for extensions as well (we could link to https://github.com/Automattic/jetpack/blob/90a1b683c84b87a401ae707599842fe59255ab9b/extensions/README.md ), and a build for Instant Search ( https://github.com/Automattic/jetpack/blob/9aee29df9f2b1ad9031eabe5d7118ad9e5a6f73c/modules/search/instant-search/README.md )

Maybe we can mention that here?

bisko replied:
Thanks! Added a section about them.

jeherve commented on docs/development-environment.md:
````diff
@@ -1,298 +1,444 @@
+	The production build will generate minified files without duplicated code (resulting from dependencies) and will also generate the matching source map and language files. To build it use:
+	
+	```sh
+	NODE_ENV=production yarn build-client
````
We can shorten that to `yarn build-production-client`.

bisko replied:
Thanks! Updated.

jeherve commented on docs/development-environment.md:
````diff
@@ -1,298 +1,444 @@
+	$ composer php:lint
+	```
+	
+	_There's also a handy `yarn php:lint` that will run `composer php:lint` if you prefer_.
````
I wonder if we should keep only the Yarn version in here to keep things shorter?

bisko replied:
Makes sense, as it will reduce clutter. Updated.

jeherve commented on docs/development-environment.md:
```diff
@@ -1,298 +1,444 @@
+	
+	More info can be found in [our support documentation](https://jetpack.com/support/jetpack-cli/).
+
+* ### JETPACK_DEV_DEBUG
```
Do we need those to be list items? Can't they just be sub-headings?

bisko replied:
The only reason I left this as a list is to indent them. It looks a bit neater to have them indented under the headline and the list seems to be the only way to achieve that.

jeherve reviewed (commented):
Thanks for tackling this! I left some remarks below.

bisko committed:
Update links to to sections using GitHub's syntax

GitHub offers different syntax for linking sections. This update removes the manual links in the content and relies on GitHub's linking mechanism.

bisko commented on docs/development-environment.md:
```diff
@@ -1,298 +1,444 @@
+* [Standard development & debugging tools](#devtools)
+
+      
+<a name="section-setting-up-env"></a>
```
Thanks! I removed the custom links and resorted to using GitHub's syntax. This was an adventure on it's own :)

bisko committed:
Update and address feedback

bisko commented on docs/development-environment.md:
```diff
@@ -1,298 +1,444 @@
+
+	VVV is similar to Docker in how it works, but instead of setting up separate containers for the different parts it uses a single Linux virtual machine to set everything up with a nice interface.
+	
+	*...This section is still under development...*
```
Thanks! Updated to include

bisko commented on docs/development-environment.md:
```diff
@@ -1,298 +1,444 @@
+
+To get a local WordPress site up and running you need a web server (Apache, Nginx), PHP and MySQL (or MariaDB).
+
+For easier and more organized development, there are several ways to achieve that:
```
Updated the text to mention that.

bisko committed:
Remove extra prompt prefixes and mark formatting as sh

bisko committed:
Add information about building the additional Jetpack modules. Fix additional things missed on previous passes.

bisko committed:
Fix typos

jeherve reviewed (dismissed):
This looks good, thanks a lot for taking a pass at this. 

It should be good to merge! 👍 🚢

bisko committed:
Fix yarn install URL, based on #15142
